### PR TITLE
Fix issue DPTP-3312 Clean up leaked resources

### PR DESCRIPTION
**Pull request by bear-redhat**
Clean up HyperShift created but expired AWS resources.

### CODE REVIEW

File: hack/cleanup-hypershift-leaks.sh [hunk 1]
```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+export NOW=$(date +%s)
```
Comment (jmguzik) [inline hack/cleanup-hypershift-leaks.sh:2]: does it need to be export?

File: hack/cleanup-hypershift-leaks.sh [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+echo Cleanning up resources older than $(date -u -d @$NOW +%Y-%m-%dT%H:%M:%SZ)
+
+echo 'Deleting expired OpenID Connect Providers...'
+# Get all OpenID Connect Providers with ARN
+arns=`aws iam list-open-id-connect-providers --query 'OpenIDConnectProviderList[*].Arn' --output json | jq -r '.[]'`
```
Comment (jmguzik) [inline hack/cleanup-hypershift-leaks.sh:7]: what other tools are needed as dependencies of this script? I would mark that somehow.

File: hack/cleanup-hypershift-leaks.sh [hunk 3]
```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
```
Comment (jmguzik) [inline hack/cleanup-hypershift-leaks.sh:1]: I would generally make an image out of this script if we would like to run it as periodic. I would move it to images/ and compose a dockerfile

**Comment by jmguzik** [comment]
See https://github.com/openshift/ci-tools/blob/master/images/quayio-pull-through-cache/Dockerfile in this repo. If you want to go this way, you need to make changes in the release as well, I can guide you through that.

**Comment by bear-redhat** [comment]
@jmguzik  I converted the tool to be a docker image. I will add the job part once its merged.

File: images/dptp-3312-hypershift-leaks-cleaner/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,7 @@
+FROM amazon/aws-cli:latest
```
Comment (jmguzik) [inline images/dptp-3312-hypershift-leaks-cleaner/Dockerfile:1]: If that is coming from dockerhub (as I think) we need to think about a different solution. It is rate-limited on our cluster. The image will not be downloaded.

**Review by jmguzik — CHANGES_REQUESTED** [review]

**Review by jmguzik — APPROVED** [review]

**Pull request merged**